lib/stdio/encoding.js

```diff
@@ -5,21 +5,22 @@ import {isUint8Array} from './utils.js';
 // Apply the `encoding` option using an implicit generator.
 // This encodes the final output of `stdout`/`stderr`.
 export const handleStreamsEncoding = (stdioStreams, {encoding}, isSync) => {
-	if (stdioStreams[0].direction === 'input' || IGNORED_ENCODINGS.has(encoding) || isSync) {
-		return stdioStreams.map(stdioStream => ({...stdioStream, encoding}));
```

Comment:
`stdioStream` represents a value with the `stdin`/`stdout`/`stderr`/`stdio` option.

`stdioStream.encoding` is set the same as the `encoding` option. It is used to know whether to convert the chunks passed to the transform as strings or as `Uint8Array`.